### PR TITLE
Fix various lint issues - part 3

### PR DESCRIPTION
**Pull request by winstonsung**
- C0103: invalid-name
- C0301: line-too-long
- C0303: trailing-whitespace
- C0325: superfluous-parens
- C0411: wrong-import-order
- R1705: no-else-return
- R1719: simplifiable-if-expression
- W1309: f-string-without-interpolation
- W1514: unspecified-encoding

Bug: #45
Change-Id: Id4ec6866e3ae003c2e7b0320b6ffe8fb6b38508b

## Summary by CodeRabbit

- **New Features**
  - Added method `setup(bot)` for initializing bot configurations.
  - Introduced `create_ctf()` function for enhanced CTF creation.

- **Bug Fixes**
  - Corrected assignment in `charge` method to ensure proper ticket management.

- **Improvements**
  - Updated SQL queries for table creation and database operations.
  - Renamed methods and parameters for better readability and consistency.

- **Refactor**
  - Modified various methods and classes for improved code structure and maintainability.

### CODE REVIEW

**Comment by Edit-Mr** [comment]
> Important
> 
> ## Auto Review Skipped
> Auto reviews are disabled on base/target branches other than the default branch. Please add the base/target branch pattern to the list of additional branches to be reviewed in the settings.
> 
> Please check the settings in the CodeRabbit UI or the `.coderabbit.yaml` file in this repository. To trigger a single review, invoke the `@coderabbitai review` command.
> 
> You can disable this status message by setting the `reviews.review_status` to `false` in the CodeRabbit configuration file.
> Thank you for using CodeRabbit. We offer it for free to the OSS community and would appreciate your support in helping us grow. If you find it useful, would you consider giving us a shout-out on your favorite social media?
> 
> Share
> Tips

Review it

**Comment by Edit-Mr** [comment]
@coderabbitai According to the code winstonsung wrote, how old do you think he is?

File: cog/ctf.py [hunk 1]
```diff
@@ -5,27 +5,25 @@
...
-import random
-import os
+from discord.ext import commands
+# Local imports
+from build.build import Build
 from cog.core.sql import read
 from cog.core.sql import write
-from cog.core.sql import end as endSQL    #用來結束和SQL資料庫的會話，平常都用end()，但和 Discord 指令變數名稱衝突，所以這裡改名
+# 用於結束和SQL資料庫的會話，平常都用end()，但和 Discord 指令變數名稱衝突，所以這裡改名
+from cog.core.sql import end as end_sql
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/ctf.py]: Done.

File: cog/comment.py [hunk 2]
```diff
@@ -172,56 +178,75 @@ async def count(message):
...
-            hexColor = ''.join([c*2 for c in hexColor])#格式化成六位數
-            embed = discord.Embed(title=f"#{hexColor}\n{correct:.2f}%", color=discord.Colour(int(hexColor, 16)))
-            await message.channel.send(embed=embed)
-            niceColor = ''.join([c*2 for c in niceColor])#格式化成六位數
-            embed = discord.Embed(description=f"答案:左邊顏色\n總共回答次數:{round}", color=discord.Colour(int(niceColor,16)))
-            await message.channel.send(embed=embed)
+            cursor.execute("UPDATE game SET niceColorRound = niceColorRound + 1;")
+            # https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/consider-using-generator.html
+            # pylint: disable-next = line-too-long
+            correct = 100 - sum((int(hex_color[i], 16) - int(nice_color[i], 16)) ** 2 for i in range(0, 3)) ** 0.5 / 0.2598076211353316
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/comment.py]: Done.

File: cog/ctf.py [hunk 3]
```diff
@@ -5,27 +5,25 @@
...
 from cog.core.sql import link_sql
-from datetime import datetime
-from build.build import Build
-
 
-def getCTFmakers():
-    with open(f"{os.getcwd()}/DataBase/server.config.json", "r") as file:
+def get_ctf_makers():
+    with open(f"{os.getcwd()}/DataBase/server.config.json", "r", encoding = "utf-8") as file:
         return json.load(file)
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/ctf.py]: Done.

**Comment by winstonsung** [comment]
@coderabbitai review

**Comment by winstonsung** [comment]
@coderabbitai review

**Comment by Edit-Mr** [comment]
@winstonsung 我該 merge 嗎

**Comment by winstonsung** [comment]
你覺得沒問題就可以合併了。還在改的會寫WIP並且標為draft（草稿）。

File: cog/ctf.py [hunk 4]
```diff
@@ -41,143 +59,182 @@ def __init__(self, *args, **kwargs) -> None:
...
-        
-        CONNECTION,CURSOR=link_sql()#SQL 會話
-        CURSOR.execute("USE CTF;")
-        while (1):
-            newId = generateCTFId()
-            #找尋是否有重複的ID，若無則跳出迴圈
-            CURSOR.execute(f"select id from data WHERE EXISTS(select id from data WHERE id={newId});")
-            idExist=CURSOR.fetchone()
-            if(idExist==None):
+        # ctfFile = get_ctf_file()
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/ctf.py:219]: Done.

File: cog/ctf.py [hunk 5]
```diff
@@ -41,143 +59,182 @@ def __init__(self, *args, **kwargs) -> None:
...
-        limit: Option(int, "限制回答次數", required=False, default=''),
-        case: Option(bool, "大小寫忽略", required=False, default=False), 
-        start: Option(str, f"開始作答日期 ({datetime.now().strftime('%y/%m/%d %H:%M:%S')})", required=False, default=""), #時間格式
-        end: Option(str, f"截止作答日期 ({datetime.now().strftime('%y/%m/%d %H:%M:%S')})", required=False, default="")):
-        #SQL沒有布林值，所以要將T/F轉換成0或1
-        case=1 if case else 0
-        role_id =getCTFmakers()["SCAICT-alpha"]["SP-role"]["CTF_Maker"]#get ctf maker role's ID 
+                    await interaction.message.edit(embed = embed)
+                    end_sql(connection, cursor) # 結束SQL會話
+            await interaction.response.send_modal(SubmitModal(title = "你找到 flag 了嗎？"))
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/ctf.py:190]: Done.

File: cog/comment.py [hunk 6]
```diff
@@ -172,56 +180,75 @@ async def count(message):
...
+            await message.channel.send(embed = embed)
         # except:
-        #     await message.add_reaction("❔")
+            # await message.add_reaction("❔")
             # print error message
-            
-        end(CONNECT,CURSOR)
-        
+
+        end(connection, cursor)
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/comment.py:251]: Done.

**Comment by winstonsung** [comment]
@coderabbitai review

File: cog/game.py [hunk 7]
```diff
@@ -1,42 +1,49 @@
...
-    @discord.slash_command(name="rock_paper_scissors", description="玩剪刀石頭布")
+    @discord.slash_command(name = "rock_paper_scissors", description = "玩剪刀石頭布")
     # useser can choose ✊, 🤚, or ✌️ in their command
-    async def rock_paper_scissors(self, interaction, choice: discord.Option(str, choices=["✊", "🤚", "✌️"])):
-        if (interaction.channel.id!=getChannels()["channel"]["commandChannel"]):
+    async def rock_paper_scissors(
+        self,
+        interaction,
+        choice: discord.Option(str, choices = [ "✊", "🤚", "✌️" ])
+    ):
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/game.py:28]: Done.

File: cog/game.py [hunk 8]
```diff
@@ -1,42 +1,49 @@
...
             await interaction.response.send_message("這裡不是指令區喔")
             return
-        userId = interaction.user.id
-        nickName = interaction.user
-        CONNECTION,CURSOR=link_sql()#SQL 會話
-        
-        point = read(userId,'point',CURSOR)
-        if point<5:
+        user_id = interaction.user.id
+        nickname = interaction.user
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/game.py:33]: Done.

File: cog/game.py [hunk 9]
```diff
@@ -1,42 +1,49 @@
...
+        connection, cursor = link_sql() # SQL 會話
+
+        point = read(user_id, 'point', cursor)
+        if point < 5:
             await interaction.response.send_message("你的電電點不足以玩這個遊戲")
-            end(CONNECTION,CURSOR)
+            end(connection, cursor)
             return
-        if choice not in ["✊", "🤚", "✌️"]:
+        if choice not in [ "✊", "🤚", "✌️" ]:
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/game.py:41]: Ignore.

File: cog/game.py [hunk 10]
```diff
@@ -46,27 +53,31 @@ async def rock_paper_scissors(self, interaction, choice: discord.Option(str, cho
...
+            print(f"{user_id}, {nickname} Get {game_outcomes[(bot_choice, choice)]} point by playing rock-paper-scissors")
+        write(user_id, 'point', point, cursor)
+        end(connection, cursor)
+
+    @discord.slash_command(name = "number_status", description = "數數狀態")
+    async def number_status(self, interaction):
+        connection, cursor = link_sql() # SQL 會話
+        cursor.execute("SELECT seq FROM game")
+        nowStatus = cursor.fetchone()[0]
+        end(connection, cursor)
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/game.py:74]: Done.

File: cog/comment.py [hunk 11]
```diff
@@ -4,56 +4,61 @@
...
 from cog.core.sql import link_sql
 
-import random
-def insert_user(userId, TABLE, CURSOR): # 初始化（新增）傳入該ID的資料表
-    CURSOR.execute(f"INSERT INTO {TABLE} (uid) VALUE({userId})") # 其他屬性在新增時MySQL會給預設值
+with open(f"{os.getcwd()}/DataBase/server.config.json", "r", encoding = "utf-8") as file:
+    stickers = json.load(file)["SCAICT-alpha"]["stickers"]
+
+def insert_user(user_id, table, cursor): # 初始化（新增）傳入該ID的資料表
+    cursor.execute(f"INSERT INTO {table} (uid) VALUE({user_id})") # 其他屬性在新增時MySQL會給預設值
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/comment.py:25]: Done.

File: cog/comment.py [hunk 12]
```diff
@@ -4,56 +4,61 @@
...
-    times=read(userId,"times",CURSOR,TABLE="CommentPoints")
+    with open(f"{os.getcwd()}/DataBase/server.config.json", "r", encoding = "utf-8") as config_file:
+        return json.load(config_file)["SCAICT-alpha"]["channel"]
+
+def reset(message, now, cursor):
+    user_id = message.author.id
+    write(user_id, "today_comments", 0, cursor) # 歸零發言次數
+    write(user_id, "last_comment", str(now), cursor)
+    write(user_id, "times", 2, cursor, table = "CommentPoints") # 初始化達標後能獲得的電電點
+    write(user_id, "next_reward", 1, cursor, table = "CommentPoints")
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/comment.py:37]: Done.

File: cog/comment.py [hunk 13]
```diff
@@ -4,56 +4,61 @@
...
-    #讀USER資料表的東西
-    userId=message.author.id
-    nickName=message.author
-    today_comments=read(userId,"today_comments",CURSOR)
-    point=read(userId,"point",CURSOR)
-    #讀CommentPoints 資料表裡面的東西，這個表格紀錄有關發言次數非線性加分的資料
-    next_reward=read(userId,"next_reward",CURSOR,TABLE="CommentPoints")
-    times=read(userId,"times",CURSOR,TABLE="CommentPoints")
+    with open(f"{os.getcwd()}/DataBase/server.config.json", "r", encoding = "utf-8") as config_file:
+        return json.load(config_file)["SCAICT-alpha"]["channel"]
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/comment.py:30]: Done.

File: cog/comment.py [hunk 14]
```diff
@@ -4,56 +4,61 @@
...
-        write(userId, "times", times, CURSOR, TABLE = "CommentPoints")
+        write(user_id, "point", point, cursor)
+        write(user_id, "next_reward", next_reward, cursor, table = "CommentPoints")
+        write(user_id, "times", times, cursor, table = "CommentPoints")
 
         # 紀錄log
-        print(f"{userId},{nickName} Get 2 point by comment {datetime.now()}")
-    write(userId, "today_comments", today_comments, CURSOR)
+        print(f"{user_id}, {nickname} Get 2 point by comment {datetime.now()}")
+    write(user_id, "today_comments", today_comments, cursor)
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/comment.py:61]: Done.

File: cog/comment.py [hunk 15]
```diff
@@ -65,163 +70,198 @@ def __init__(self, bot):
...
+                # 猜色碼回應
+                    await Comment.nice_color(message)
+                if message.channel.id not in self.sp_channel["exclude_point"]:
+                    # 平方發言加電電點，列表中頻道不算發言次數
+                    Comment.today_comment(user_id, message, cursor)
+        # pylint: disable-next = broad-exception-caught
+        except Exception as exception:
+            print(f"Error: {exception}")
+
+        end(connection, cursor)
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/comment.py:99]: Ignore.

File: cog/comment.py [hunk 16]
```diff
@@ -65,163 +70,198 @@ def __init__(self, bot):
...
+                if message.channel.id == self.sp_channel["countChannel"]:
+                # 數數回應
+                    await Comment.count(message)
+                elif message.channel.id == self.sp_channel["colorChannel"]:
+                # 猜色碼回應
+                    await Comment.nice_color(message)
+                if message.channel.id not in self.sp_channel["exclude_point"]:
+                    # 平方發言加電電點，列表中頻道不算發言次數
+                    Comment.today_comment(user_id, message, cursor)
+        # pylint: disable-next = broad-exception-caught
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/comment.py:95]: Done.

File: cog/comment.py [hunk 17]
```diff
@@ -65,163 +70,198 @@ def __init__(self, bot):
...
         # 新增該user的資料表
-        if not user_id_exists(userId, "USER", CURSOR): # 該 uesr id 不在USER資料表內，插入該筆使用者資料
-            insert_user(userId, "USER", CURSOR)
-        if not user_id_exists(userId, "CommentPoints", CURSOR):
-            insert_user(userId, "CommentPoints", CURSOR)
+        if not user_id_exists(user_id, "USER", cursor):
+            # 該 uesr id 不在USER資料表內，插入該筆使用者資料
+            insert_user(user_id, "USER", cursor)
+        if not user_id_exists(user_id, "CommentPoints", cursor):
+            insert_user(user_id, "CommentPoints", cursor)
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/comment.py:108]: Done.

File: cog/comment.py [hunk 18]
```diff
@@ -65,163 +70,198 @@ def __init__(self, bot):
...
+                    await message.add_reaction("❌")
+                    await message.add_reaction("❓")
+            except (TypeError, ValueError):
+                # 在decimal_number賦值因為不是數字（可能聊天或其他文字）產生錯誤產生問號emoji回應
+                await message.add_reaction("❔")
+        # pylint: disable-next = broad-exception-caught
+        except Exception as exception:
+            print(f"Error: {exception}")
+
+        end(connection, cursor)
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/comment.py:193]: Done.

File: cog/comment.py [hunk 19]
```diff
@@ -65,163 +70,198 @@ def __init__(self, bot):
...
+                )
+                await message.channel.send(embed = embed)
+            # except:
+                # await message.add_reaction("❔")
+                # print error message
+        # pylint: disable-next = broad-exception-caught
+        except Exception as exception:
+            print(f"Error: {exception}")
+
+        end(connection, cursor)
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/comment.py:264]: Ignore.

File: cog/comment.py [hunk 20]
```diff
@@ -65,163 +70,198 @@ def __init__(self, bot):
...
+                        write(message.author.id, "point", point, cursor)
+                        # pylint: disable-next = line-too-long
+                        print(f"{message.author.id}, {message.author} Get 5 point by count reward {datetime.now()}")
+                        await message.add_reaction("💸")
+                else:
+                    # 不同人數數，但數字不對
+                    await message.add_reaction("❌")
+                    await message.add_reaction("❓")
+            except (TypeError, ValueError):
+                # 在decimal_number賦值因為不是數字（可能聊天或其他文字）產生錯誤產生問號emoji回應
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply cog/comment.py:187]: Ignore.

**Pull request merged**